Sources/_Datadog_Private/ObjcAppLaunchHandler.m

```diff
@@ -56,32 +57,60 @@ @interface AppLaunchHandler : NSObject
 @implementation AppLaunchHandler
 
 + (void)load {
+    pthread_rwlock_init(&rwLock, NULL);
+
     // This is called at the `_Datadog_Private` load time, keep the work minimal
     FrameworkLoadTime = CFAbsoluteTimeGetCurrent();
 
     NSNotificationCenter * __weak center = NSNotificationCenter.defaultCenter;
-    id __block token = [center
-                        addObserverForName:UIApplicationDidBecomeActiveNotification
-                        object:nil
-                        queue:NSOperationQueue.mainQueue
-                        usingBlock:^(NSNotification *_){
-
-        pthread_rwlock_init(&rwLock, NULL);
+    id __weak __block token = [center addObserverForName:UIApplicationDidBecomeActiveNotification
```

Comment:
if this is `__weak`, who retains `token`? i guess this may crash

Reply:
It would crash systematically [here](https://github.com/DataDog/dd-sdk-ios/pull/730/files#diff-12e60ffaca99dbb26bf5cb6c41b8b78575a811012b4eea1574d9fb07c7254279R73). It's retained by the notification center, from apple doc: 
> Notification center strongly holds this return value until you remove the observer registration.